packages/next/build/webpack/plugins/build-manifest-plugin.ts

```diff
@@ -12,19 +12,43 @@ import {
 import { BuildManifest } from '../../../next-server/server/get-page-files'
 import getRouteFromEntrypoint from '../../../next-server/server/get-route-from-entrypoint'
 import { ampFirstEntryNamesMap } from './next-drop-client-page-plugin'
+import { Rewrite } from '../../../lib/load-custom-routes'
+import {
+  isDynamicRoute,
+  getSortedRoutes,
+} from '../../../next-server/lib/router/utils'
 
 const isWebpack5 = parseInt(webpack.version!) === 5
 
 // This function takes the asset map generated in BuildManifestPlugin and creates a
 // reduced version to send to the client.
 function generateClientManifest(
   assetMap: BuildManifest,
-  isModern: boolean
+  isModern: boolean,
+  rewrites: Rewrite[]
 ): string {
-  const clientManifest: { [s: string]: string[] } = {}
+  const clientManifest: { [s: string]: string[] | Rewrite[] } = {
+    __rewrites: rewrites,
+  }
   const appDependencies = new Set(assetMap.pages['/_app'])
 
-  Object.entries(assetMap.pages).forEach(([page, dependencies]) => {
+  const pageKeys = Object.keys(assetMap.pages)
+  const dynamicPages: string[] = []
+
+  const sortedRoutes = [
+    ...pageKeys.filter((page) => {
+      if (isDynamicRoute(page)) {
+        dynamicPages.push(page)
+        return false
+      }
+      return true
+    }),
+    ...getSortedRoutes(dynamicPages),
+  ]
```

Comment:
`getSortedRoutes` can accept non-dynamic pages. This should simply read the following IMO:
```suggestion
  const sortedRoutes = getSortedRoutes(pageKeys)
```

Reply:
Much better 😅